apps/studio/src/server/modules/user/user.service.ts

```diff
@@ -11,3 +21,144 @@ export const isUserDeleted = async (email: string) => {
 
   return user?.deletedAt ? true : false
 }
+
+export const validateEmailRoleCombination = ({
+  email,
+  role,
+}: {
+  email: string
+  role: ResourcePermission["role"]
+}) => {
+  if (!isGovEmail(email) && role === RoleType.Admin) {
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message:
+        "Non-gov.sg emails cannot be added as admin. Select another role.",
+    })
+  }
+}
+
+interface CreateUserProps {
+  email: User["email"]
+  name?: User["name"]
+  phone?: User["phone"]
+  role: ResourcePermission["role"]
+  siteId: ResourcePermission["siteId"]
+  trx?: SafeKysely // allows for transaction to be passed in from parent transaction
+}
+
+export const createUser = async ({
```

Comment:
[nit] this name is abit misleading because the method also adds permissions for the user; what do you feel about `createUserWithPermissionsForSite` (can shorten but you get the idea)

Reply:
good point, updated in https://github.com/opengovsg/isomer/pull/1118/commits/82e1e85e6a2a354171efea8cd57adb92f33c036f